src/services/consent/hooks/consents.js

```diff
@@ -1,272 +1,8 @@
-	hook.result = (hook.result.constructor.name === 'model') ? hook.result.toObject() : hook.result;
-	const consentPromises = (hook.result.data || [])
-		.map((consent) => accessCheck(consent, hook.app)
-			.then((result) => result));
-
-	return Promise.all(consentPromises).then((users) => {
-		hook.result.data = users;
-		return Promise.resolve(hook);
-	});
-};
-
-// this method is currently duplicated in AdminUsers service
-const getConsentStatus = (consent) => {
-	const isUserConsent = (c = {}) => {
-		const uC = c.userConsent;
-		return uC && uC.privacyConsent && uC.termsOfUseConsent;
-	};
-
-	const isNOTparentConsent = (c = {}) => {
-		const pCs = c.parentConsents || [];
-		return pCs.length === 0 || !(pCs.some((pC) => pC.privacyConsent && pC.termsOfUseConsent));
-	};
-
-	if (consent.requiresParentConsent) {
-		if (isNOTparentConsent(consent)) {
-			return 'missing';
-		}
-
-		if (isUserConsent(consent)) {
-			return 'ok';
-		}
-		return 'parentsAgreed';
-	}
-
-	if (isUserConsent(consent)) {
-		return 'ok';
-	}
-
-	return 'missing';
-};
-
-const addConsentStatus = (hook) => {
-	if (hook.result) {
-		hook.result.consentStatus = getConsentStatus(hook.result);
+exports.restrictToCurrentUser = (context) => {
+	if (context.id === context.params.authentication.payload.userId
+		|| context.params.query.userId === context.params.authentication.payload.userId) {
+		return context;
 	}
-};
-
-const addConsentsStatus = (hook) => {
-	if (hook.result.data) {
-		hook.result.data.forEach((consent) => {
-			consent.consentStatus = getConsentStatus(consent);
-		});
-	}
-};
-
-// TODO: remove at next version
-const writeConsentToUser = (context) => {
-	const { data, app } = context;
-	app.service('usersModel').patch(data.userId, modifyDataForUserSchema(data));
-};
-
-const patchConsentToUser = async (context) => {
-	const { data, app, id } = context;
-	const consent = await app.service('consents').get(id);
-	app.service('usersModel').patch(consent.userId, modifyDataForUserSchema(data));
-};
-
-const removeConsentFromUser = async (context) => {
-	const { app, id } = context;
-	const consent = await app.service('consents').get(id);
-	const user = await app.service('usersModel').get(consent.userId);
-	delete user.consent;
-	await app.service('usersModel').update(user.id, user);
-};
-
-exports.before = {
-	all: [],
-	find: [
-		authenticate('jwt'),
-		globalHooks.ifNotLocal(restrictToUserOrRole),
-		setUserIdToCorrectForm,
-	],
-	get: [authenticate('jwt')],
-	create: [addDates, checkExisting, writeConsentToUser],
-	update: [authenticate('jwt'), addDates, writeConsentToUser],
-	patch: [authenticate('jwt'), addDates, patchConsentToUser],
-	remove: [authenticate('jwt'), removeConsentFromUser],
-};
 
-exports.after = {
-	all: [],
-	find: [decorateConsents, addConsentsStatus],
-	get: [decorateConsent, addConsentStatus],
-	create: [],
-	update: [],
-	patch: [],
-	remove: [],
+	throw new Forbidden('You are only allowed to request your onw consent');
```

Comment:
Codacy found an issue: ['Forbidden' is not defined.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5680233)